src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -1166,8 +1095,61 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
 
     }
 
-Done:
-    ;
+    HELPER_METHOD_FRAME_END();
+
+    return OBJECTREFToObject(gc.retVal);
+}
+FCIMPLEND
+
+
+FCIMPL2(Object*, RuntimeMethodHandle::InvokeArrayCtor,
+    Span<OBJECTREF>* objs,
+    SignatureNative* pSigUNSAFE)
+{
+    FCALL_CONTRACT;
+
+    struct
+    {
+        SIGNATURENATIVEREF pSig;
+        OBJECTREF retVal;
+    } gc;
+
+    gc.pSig = (SIGNATURENATIVEREF)pSigUNSAFE;
```

Comment:
The old code also had a check for `IsSharedByGenericInstantiations`. We may need it here as well.

(In theory, it should not possible to get a managed Type for `IsSharedByGenericInstantiations` type. In practice, we always had odd corner cases or bugs how one can end up with them.)

Reply:
Yes. I did check that from the typical case and it trigger long before it got here. I can add it back here if you think it is warranted.

Reply:
I would either add it here or delete it from the other place so that things are consistent.